tools/sof-fw-boot-test.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import subprocess
+import sys
+import time
+from pathlib import Path
+
+# define command line arguments
+def parse_cmdline():
+	parser = argparse.ArgumentParser(add_help=True, formatter_class=argparse.RawTextHelpFormatter,
+	    description='A script for stress testing firmware boot')
+
+	parser.add_argument('-i', '--iter', type=int, default=100, help="number of firmware boot iterations")
+	parser.add_argument('-f', '--firmware', type=str, required=True, help='firmware filename')
+	parser.add_argument('-p', '--fw_path', type=str, required=True, help='path to the firmware file relative to /lib/firmware')
+
+	return vars(parser.parse_args())
+
+def main():
+	cmd_args = parse_cmdline()
+
+	# Get firmware file path
+	fw_path = cmd_args["fw_path"]
+	full_fw_path = '/lib/firmware/' + fw_path
+	if not os.path.exists(full_fw_path):
+		print(f"Firmware path {fw_path} does not exist")
+		sys.exit(1)
+
+	# Get firmware file name
+	fw_filename = cmd_args["firmware"]
+
+	# check if the firmware binary exists
+	fw_fullname = '/lib/firmware/' + fw_path + '/' + fw_filename
+	if not os.path.isfile(fw_fullname):
+		print(f"Firmware file {fw_filename} does not exist")
+		sys.exit(1)
+
+	num_iter = cmd_args["iter"]
+	output = f"""==============================================================================
+	Starting boot stress test with:
+	Firmware filename: {fw_filename}
+	Path to firmware file: {fw_path}
+	Number of Iterations: {num_iter}
+=============================================================================="""
+	print(output)
+
+	# set path to the depbugfs entries
+	debugfs_path = '/sys/kernel/debug/sof/dsp_ops'
+	if not os.path.exists(debugfs_path):
+		print(f"Debugfs path {debugfs_path} does not exist!")
+		sys.exit(1)
+
+	# set the firmware filename & path
+	os.system('echo %s > %s/fw_filename' %(fw_filename, debugfs_path))
+	os.system('echo %s > %s/fw_path' %(fw_path, debugfs_path))
```

Comment:
You really don't need to mix python and shell script like this. Just stick to Python (TESTED), it's faster and you will get better error messages:

```python
   open(debugfs_path + '/fw_filename', 'w').write(fw_filename)
   open(debugfs_path + '/fw_path', 'w').write(fw_path)
```

Reply:
I had to add a \n to the fw_filename for this to work as intened. Otherwise the last character in the fw_filename was getting dropped